### PR TITLE
Replace some deprecated functions

### PR DESCRIPTION
**Pull request by andreasfelix**
This gets rid of some warning messages during compilation.

### CODE REVIEW

File: src/MainWindow.vala [hunk 1]
```diff
@@ -331,7 +331,7 @@ namespace Terminal {
         protected bool match_keycode (int keyval, uint code) {
 #endif
             Gdk.KeymapKey [] keys;
-            Gdk.Keymap keymap = Gdk.Keymap.get_default ();
+            Gdk.Keymap keymap = Gdk.Keymap.get_for_display (get_display ());
```
Comment (jeremypw) [inline src/MainWindow.vala:334]: Might as well follow current style and use `var keymap = ...` here.

File: src/MainWindow.vala [hunk 2]
```diff
@@ -726,8 +726,7 @@ namespace Terminal {
             default_height = rect.height;
 
             if (default_width == -1 || default_height == -1) {
-                Gdk.Rectangle geometry;
-                get_screen ().get_monitor_geometry (get_screen ().get_primary_monitor (), out geometry);
+                Gdk.Rectangle geometry = get_display ().get_primary_monitor ().get_geometry ();
```
Comment (jeremypw) [inline src/MainWindow.vala:729]: Might as well use `var geometry = .....` here.

**Review by jeremypw — CHANGES_REQUESTED** [review]
Good cleanup. Just a couple of very minor comments.

File: src/MainWindow.vala (hunk 1 above)
Comment (andreasfelix) [reply src/MainWindow.vala:334]: done

File: src/MainWindow.vala (hunk 2 above)
Comment (andreasfelix) [reply src/MainWindow.vala]: done

**Comment by andreasfelix** [comment]
Is it save to remove this line? As far as I understand [this setting is ignored now](https://developer.gnome.org/gtk3/stable/GtkSettings.html#GtkSettings--gtk-menu-bar-accel).

https://github.com/elementary/terminal/blob/7c970414b8a811e2985ec6a0d2e9ebc289ae0fe3/src/MainWindow.vala#L191-L193

**Comment by jeremypw** [comment]
I would stick to the stated scope of the PR tbh if you want a quick merge.

**Comment by jeremypw** [comment]
Have you looked into replacing the `spawn_sync ()` function? (see my comment on Slack). I couldn't get `spawn_async ()` to work - looks like the `.vapi` is wrong.

**Comment by andreasfelix** [comment]
> I would stick to the stated scope of the PR tbh if you want a quick merge.

The `gtk_settings.gtk_menu_bar_accel` also throw a deprecation warning, therefore I considered it to be the same scope.

**Comment by andreasfelix** [comment]
> Have you looked into replacing the `spawn_sync ()` function? (see my comment on Slack). I couldn't get `spawn_async ()` to work - looks like the `.vapi` is wrong.

tbh I intentionally left that out because I haven't used async in vala before. It works a bit different than in python, js or rust, the languages I happen to know :D

I am only in the community slack. I suppose there is another one, but I am not part of it. So I couldn't see you comment.

**Comment by danrabbit** [comment]
@andreasfelix I would be happy to send you an invite to the contributor slack. If you want to DM me on the community slack or have an email address I can send an invite link to. I can't seem to find you on the community slack

**Review by jeremypw — APPROVED** [review]
Removing the gtk_menu_bar_accel override should be safe enough I guess - it isn't used elsewhere in elementary or within Terminal.  I guess it is to be sure that terminal programs like `htop` can be exited with it.

**Pull request merged**